github-pullrequest-plugin/src/main/java/com/github/kostyasha/github/integration/branch/trigger/check/BranchToCauseConverter.java

```diff
@@ -28,61 +27,72 @@
+
+        String name = remoteBranch.getName();
+        if (causes.isEmpty()) {
+            LOGGER.debug("No build trigger causes found for branch [{}].", name);
+            return null;
+        }
 
-        final GitHubBranchCause cause = from(trigger.getEvents())
-                .transform(toCause(remoteBranch))
-                .filter(notNull())
-                .first()
-                .orNull();
+        LOGGER.debug("All matched events for branch [{}] : {}.", name, causes);
 
-        if (cause == null || cause.isSkip()) {
+        GitHubBranchCause cause = skipTrigger(causes);
+        if (cause != null) {
+            listener.info("Build of branch [{}] skipped: {}.", name, cause.getReason());
             return null;
-        } else {
-            return cause;
         }
-    }
 
+        // use the first cause from the list
+        cause = causes.get(0);
+        LOGGER.debug("Using build cause [{}] as trigger for branch [{}].", cause, name);
 
-    @VisibleForTesting
-    /* package */ EventToCauseConverter toCause(GHBranch ghBranch) {
-        return new EventToCauseConverter(ghBranch);
+        return cause;
     }
 
-    @VisibleForTesting
-    /* package */ class EventToCauseConverter implements Function<GitHubBranchEvent, GitHubBranchCause> {
-        private final GHBranch remoteBranch;
+    private GitHubBranchCause skipTrigger(List<GitHubBranchCause> causes) {
+        GitHubBranchCause cause = causes.stream()
+                .filter(GitHubBranchCause::isSkip)
+                .findFirst()
+                .orElse(null);
 
-        EventToCauseConverter(GHBranch remoteBranch) {
-            this.remoteBranch = remoteBranch;
+        if (cause == null) {
+            return null;
         }
 
-        @Override
-        public GitHubBranchCause apply(GitHubBranchEvent event) {
-            //null if local not existed before
-            final GitHubBranch localBranch = localRepo.getBranches().get(remoteBranch.getName());
-            try {
-                return event.check(trigger, remoteBranch, localBranch, localRepo, listener);
-            } catch (IOException e) {
-                LOGGER.warn("Can't check trigger event", e);
-                listener.error("Can't check trigger event, so skipping branch: {}", e);
-                return null;
-            }
-        }
+        LOGGER.debug("Cause [{}] indicated build should be skipped.", cause);
+        return cause;
     }
 
+    private GitHubBranchCause toCause(GitHubBranchEvent event, GHBranch remoteBranch) {
+        String branchName = remoteBranch.getName();
+        GitHubBranch localBranch = localRepo.getBranches().get(branchName);
+
+        try {
+            return event.check(trigger, remoteBranch, localBranch, localRepo, listener);
+        } catch (IOException e) {
+            LOGGER.error("Event check failed, skipping branch [{}].", branchName, e);
+            listener.error("Event check failed, skipping branch [{}] {}", branchName, e);
```

Comment:
1. skip event
2. trigger event

If 1. failed and was skipped then 2. will trigger build that could cause issues if 1. was critical to not run. In such case the whole branch check should be skipped.